telethon/telegram_client.py

```diff
@@ -476,6 +476,55 @@ def get_message_history(self,
 
         return total_messages, result.messages, entities
 
+    def get_all_participants(self,entity):
+        """
+        Returns all the partcipants for the specified entity.
+
+        :param entity:      The entity from whom to retrieve the participants
+
+        :return: A list of participants for the given entity
+        """
+
+        def update_participant_set(participants, participant):
+            """
+            Updates the set of participants for the specified entity.
+
+            :param participants:    Set of participants in the chat or group
+            :param participant:     Participant in the chat or group.
+
+            :return: void
+            """
+            if participant in participants:
+                participants.remove(participant)
+            else:
+                participants.add(participant)
+
+        entity_type = self.get_entity(entity)
+        if isinstance(entity_type,User):
```

Comment:
Just to be on the safe side, you should also handle `UserEmpty` here. `if isinstance(entity_type, (User, UserEmpty))`.

By the way, override `entity = self.get_entity(entity)` so whatever calls use this doesn't have to call it again.

Reply:
Cool will make the changes!